src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -203,4 +204,69 @@ public void testParseFilePermissionChange() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testParseLineChangesAddLine() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            Files.write(helloWorld.toPath(), "line 0\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "add original line");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            Files.write(helloWorld.toPath(), "line 1 added\nline 2 added\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "add line 1, 2");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals("There should be 1 change detected", 1, changes.size());
+            final GitChange expected = ImmutableGitChange.builder()
+                    .path("HelloWorld")
+                    .addLines(1, 2)
+                    .build();
+            assertEquals("The change is not as expected", expected, changes.get(0));
+        }
+    }
+
+    @Test
+    public void testParseLineChangesRemoveLine() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            Files.write(helloWorld.toPath(), "line 0\nline 1 to be removed\nline 2 to be removed\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "add original three lines");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            Files.write(helloWorld.toPath(), "line 0\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.TRUNCATE_EXISTING);
+            GitUtils.addAllAndCommit(repository, "remove two lines");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals("There should be 1 change detected", 1, changes.size());
+            final GitChange expected = ImmutableGitChange.builder()
```

Comment:
As new test shows, removed lines are completely invisible from view.
This can stay for now, but I think it has to change as it will cause issues for some types of regression.

How will we know what method/code/comment was changed if no lines are displayed?
Issues that will be affected:
https://github.com/checkstyle/regression-tool/issues/6 - need to know which method
https://github.com/checkstyle/regression-tool/issues/49 - need to know if has changed besides chmod
https://github.com/checkstyle/regression-tool/issues/69 - need to know if was comment or code

I think we have to work with both sides of the file and not just the latest file.
@Luolc Do you agree? If so, let's create a new issue, unless you want to try and fix it now.